### ISSUE TITLE
Suggest field names when nested table type is inferred

### ISSUE THREAD

**Issue by georgiy-belyanin** [EmmyLuaLs/emmylua-analyzer-rust#502]
There is basic auto-completion suggesting table field names if the table type is known. Though it's really basic. Take this example.
```lua
---@param a { foo: { bar: number } }
function buz(a) end

buz({
    foo = { -- `foo` is suggested
        bar = 0 -- `bar` is not
    }
})
```

`a` is a nested structure, meanwhile `foo` is suggested by the language server, `bar` is not. It might be handy to have such suggestions.

**Comment by xuhuanzy** [comment]
Is it not possible to prompt for completion now?
![Image](https://github.com/user-attachments/assets/9e3ea9fe-dc5a-441c-ad44-588a5dee566a)

**Comment by xuhuanzy** [comment]
Isn't there a prompt available?
![Image](https://github.com/user-attachments/assets/44d7eb8f-5f32-4d24-aa8e-ccf57446c0f1)

**Comment by georgiy-belyanin** [comment]
> Isn't there a prompt available? ![Image](https://github.com/user-attachments/assets/44d7eb8f-5f32-4d24-aa8e-ccf57446c0f1)

It is really strange but I don't have such a suggestion. Using EmmyLua VS Code 0.9.21.

<img width="547" alt="Image" src="https://github.com/user-attachments/assets/87995c91-c621-4567-bde3-7620d742de9a" />

**Comment by xuhuanzy** [comment]
There is indeed no auto-completion when entering characters, but it works fine when auto-completion is triggered without any input. This issue will be fixed later.
